### ISSUE TITLE
Returning errors instead of throwing?

### ISSUE THREAD

**Issue by lonix1** [GetoXs/MediatR.Extensions.FluentValidation.AspNetCore#9]
Using [exceptions](https://github.com/GetoXs/MediatR.Extensions.FluentValidation.AspNetCore/blob/8d2d40078860387a8751b6e0e1ee9b702ce139ab/src/MediatR.Extensions.FluentValidation.AspNetCore/ValidationBehavior.cs#L32) for flow control is very "heavy".

Anyone figure out how to return a result with errors, instead of throwing exceptions?

**Comment by GetoXs** [comment]
You are right. But do you know any other way to pass validation failure to exception handler on middleware level?

**Comment by lonix1** [comment]
Yes it can be done using a little bit of [reflection](https://medium.com/the-cloud-builders-guild/validation-without-exceptions-using-a-mediatr-pipeline-behavior-278f124836dc). It feels "dirty", but it's still faster than using exceptions. (Doing it that way there is no need for exception handlers, the conversion from validation failure to HTTP response can be done in the controller.)

However I still hope there is another way, without exceptions and reflection.

(Feel free to close. Take a look at that link it's excellent.)
